User basic and upsert testcases for Vtgate3

saurabh408 opened the pull request:
User and Upsert VTGATE3 test cases:
basic upsert/insert ignore
sequence/sensitivity/timeout/warning
scatter updates/deletes

saurabh408 committed:
go testcases for upsert and users basic

Signed-off-by: saurabh <saurabh.408@gmail.com>

saurabh408 committed:
addressed review comments

Signed-off-by: saurabh <saurabh.408@gmail.com>

saurabh408 committed:
Merge branch 'tal_vtgate' into tal_vtgate_user_upsert

saurabh408 committed:
multi-column vindexes vtgate testcases in go

Signed-off-by: saurabh <saurabh.408@gmail.com>

deepthi commented on go/vt/vtgate/endtoend/upsert_test.go:
```diff
@@ -0,0 +1,64 @@
+	// Create lookup entries for primary vindex:
+	// No entry for 2. upsert_primary is not owned.
+	// So, we need to pre-create entries that the
+	// subsequent will insert will use to compute the
```
remove the first `will`.

deepthi reviewed (approved):
Looks good except for one doc comment change. You can merge after fixing that.
I did notice that there are no tests for upsert_owned either here or in the original python test.
They need to be added.

saurabh408 committed:
added upsert_owned testcases

Signed-off-by: saurabh <saurabh.408@gmail.com>

saurabh408 committed:
testcase for failed insert in upsert

Signed-off-by: saurabh <saurabh.408@gmail.com>

saurabh408 commented:
Deepthi, Added upsert_owned check in exiting upsert testcases also added one scenario where we got failed insert if all vindexes of upsert table leads to different keyspace-id for a row.
